lib/ops/opsservice/agents.go

```diff
@@ -581,6 +566,47 @@ func (r *AgentPeerStore) addGroup(key ops.SiteOperationKey) (*agentGroup, error)
 	return agentGroup, nil
 }
 
+func (r *AgentPeerStore) isPartOfActiveOperation(addr string, token storage.ProvisioningToken) error {
+	op, err := r.backend.GetSiteOperation(token.SiteDomain, token.OperationID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	operation := (ops.SiteOperation)(*op)
+	if operation.IsCompleted() {
```

Comment:
Is this going to play well with the fact that we always set install operation to completed when transferring it to etcd even if install is still going?